src/StreamJsonRpc.Tests/JsonRpcClient20InteropTests.cs

```diff
@@ -149,6 +150,147 @@ public async Task InvokeWithParameterPassedAsObjectAsync_NoParameter()
         Assert.Null(request["params"]);
     }
 
+    [Fact]
+    public async Task InvokeWithProgressParameterAsArray()
+    {
+        int sum = 0;
+        ProgressWithCompletion<int> progress = new ProgressWithCompletion<int>(report =>
+        {
+            sum += report;
+        });
+
+        int n = 3;
+        Task<int> invokeTask = this.clientRpc.InvokeAsync<int>("test", new object[] { n, progress });
+
+        JToken request = await this.ReceiveAsync();
+
+        long progressID = request["params"][1].Value<long>();
+
+        // Send responses as $/progress
+        int sum2 = 0;
+        for (int i = 0; i < n; i++)
+        {
+            string content = "{ \"jsonrpc\": \"2.0\", \"method\": \"$/progress\", \"params\": { \"token\": " + progressID + ", \"value\": " + i + ", } ,}";
```

Comment:
>,  [](start = 147, length = 2)

Formal JSON should not have trailing commas. The fact that newtonsoft.json accepts them is configurable, and probably not something you mean to be testing in this. i.e. we wouldn't want *this* test to start failing if we changed streamjsonrpc to stop accepting trailing commas.